src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -1,188 +1,192 @@
+class TextAnalyticsLanguageDetection(override val textAnalyticsOptions: Option[TextAnalyticsRequestOptionsV4] = None,
                                      override val uid: String = randomUID("TextAnalyticsLanguageDetection"))
   extends TextAnalyticsSDKBase[DetectedLanguageV4](textAnalyticsOptions)
-  with HasConfidenceScoreCol {
+    with HasConfidenceScoreCol {
   logClass()
 
-  /**
-   * Params for optional input column names.
-   */
-  // Country Hint
-  final val countryHintCol: Param[String] = new Param[String](this, "countryHintCol", "country hint column name")
+  override def outputSchema: StructType = DetectLanguageResponseV4.schema
+
+  override val responseTypeBinding: SparkBindings[TAResponseV4[DetectedLanguageV4]] = DetectLanguageResponseV4
 
-  final def getCountryHintCol: String = $(countryHintCol)
+  override def invokeTextAnalytics(input: Seq[String], hints: Seq[String]): TAResponseV4[DetectedLanguageV4] = {
+    val r = scala.util.Random
+    var documents = (input, hints).zipped.map { (inp, hint) =>
+      new DetectLanguageInput(r.nextInt.abs.toString, inp, hint)}.asJava
 
-  final def setCountryHintCol(value: String): this.type = set(countryHintCol, value)
-  setDefault(countryHintCol -> "CountryHint")
+    val resultCollection = textAnalyticsClient.detectLanguageBatchWithResponse(documents,
+      null,Context.NONE).getValue
 
-  override def outputSchema: StructType = DetectLanguageResponseV4.schema
+    val detectLanguageResultCollection = resultCollection.asScala
+
+    val detectLanguageResult = detectLanguageResultCollection.map(langs => langs.getPrimaryLanguage)
 
-  override protected val invokeTextAnalytics: String => TAResponseV4[DetectedLanguageV4] = (text: String) =>
-    {
-      val detectLanguageResultCollection = textAnalyticsClient.detectLanguageBatch(
-        Seq(text).asJava, null, textAnalyticsOptions.orNull)
-      val detectLanguageResult = detectLanguageResultCollection.asScala.head
-
-      val languageResult = if (detectLanguageResult.isError) {
-        None
-      } else {
-        Some(DetectedLanguageV4(
-          detectLanguageResult.getPrimaryLanguage.getName,
-          detectLanguageResult.getPrimaryLanguage.getIso6391Name,
-          detectLanguageResult.getPrimaryLanguage.getConfidenceScore))
-      }
-
-      val error = if (detectLanguageResult.isError) {
-        val error = detectLanguageResult.getError
-        Some(TAErrorV4(error.getErrorCode.toString, error.getMessage, error.getTarget))
-      } else {
-        None
-      }
-
-      val stats = Option(detectLanguageResult.getStatistics) match {
-        case Some(s) => Some(DocumentStatistics(s.getCharacterCount, s.getTransactionCount))
-        case None => None
-      }
-
-      TAResponseV4[DetectedLanguageV4](
-        languageResult,
-        error,
-        stats,
-        Some(detectLanguageResultCollection.getModelVersion))
+    val languageResult = if (detectLanguageResultCollection.head.isError) {

```

Comment:
if the first response from this batch contains errors, then the whole languageResult will be set as None?